harvester_e2e_tests/apis/test_volumes.py

```diff
@@ -103,6 +105,92 @@ def test_create_volume_using_image_by_yaml(admin_session,
         resp.content)
 
 
+def test_create_volume_with_label(admin_session, harvester_api_endpoints):
+    request_json = utils.get_json_object_from_template('basic_volume')
+    request_json['metadata']['labels'] = {
+        'test.harvesterhci.io': 'for-test'
+    }
+    resp = admin_session.post(harvester_api_endpoints.create_volume,
+                              json=request_json)
+    assert resp.status_code == 201, (
+        'Failed to create volume with labels: %s' % (resp.content))
+    view_endpoint = harvester_api_endpoints.get_volume % (
+        request_json['metadata']['name'])
+    (status, vol_data) = validate_blank_volumes(admin_session, view_endpoint)
+    assert status
+    assert vol_data['metadata']['labels'].get(
+        'test.harvesterhci.io') == 'for-test'
+    resp = admin_session.delete(harvester_api_endpoints.delete_volume % (
+        request_json['metadata']['name']))
+    # FIXME(gyee): we need to figure out why the API can arbitarily return
+    # 200 or 204. It should consistently returning one.
+    assert resp.status_code in [200, 204], 'Failed to delete volume: %s' % (
+        resp.content)
+
+
+def test_create_volume_with_image_label(admin_session,
+                                        harvester_api_endpoints, image):
+    request_json = utils.get_json_object_from_template('basic_volume')
```

Comment:
This seem like a good candidate for a fixture. Can we create a `volume_with_image` fixture so it can be shared in the future?